qds_sdk/cluster.py

```diff
@@ -174,35 +174,79 @@ def _parse_create_update(cls, args, action):
                                help="vpc to create the cluster in",)
 
         hadoop_group = argparser.add_argument_group("hadoop settings")
-        hadoop_group.add_argument("--master-instance-type",
+        node_config_group = argparser.add_argument_group("node configuration") if (api_version >= 1.3) else hadoop_group
+
+        node_config_group.add_argument("--master-instance-type",
                                   dest="master_instance_type",
                                   help="instance type to use for the hadoop" +
                                        " master node",)
-        hadoop_group.add_argument("--slave-instance-type",
+        node_config_group.add_argument("--slave-instance-type",
                                   dest="slave_instance_type",
                                   help="instance type to use for the hadoop" +
                                        " slave nodes",)
-        hadoop_group.add_argument("--initial-nodes",
+        node_config_group.add_argument("--initial-nodes",
                                   dest="initial_nodes",
                                   type=int,
                                   help="number of nodes to start the" +
                                        " cluster with",)
-        hadoop_group.add_argument("--max-nodes",
+        node_config_group.add_argument("--max-nodes",
                                   dest="max_nodes",
                                   type=int,
                                   help="maximum number of nodes the cluster" +
                                        " may be auto-scaled up to")
+        node_config_group.add_argument("--slave-request-type",
+                                  dest="slave_request_type",
+                                  choices=["ondemand", "spot", "hybrid"],
+                                  help="purchasing option for slave instaces",)
         hadoop_group.add_argument("--custom-config",
                                   dest="custom_config_file",
                                   help="location of file containg custom" +
                                        " hadoop configuration overrides")
-        hadoop_group.add_argument("--slave-request-type",
-                                  dest="slave_request_type",
-                                  choices=["ondemand", "spot", "hybrid"],
-                                  help="purchasing option for slave instaces",)
         hadoop_group.add_argument("--use-hbase", dest="use_hbase",
                                   action="store_true", default=None,
                                   help="Use hbase on this cluster",)
+        if api_version >= 1.3:
```

Comment:
Oh looks like these are for hadoop_group/node_config_group and that's why they are here. It's fine then.